src/com/ms/silverking/cloud/skfs/dir/serverside/DirectoryServer.java

```diff
@@ -46,416 +47,439 @@
-                _mode = Mode.valueOf(modeValue);
-            } catch (Exception e) {
-                Log.logErrorWarning(e, "Using default mode "+ defaultMode);
-                _mode = defaultMode;
-            }
+  private SSNamespaceStore nsStore;
+  private final ConcurrentMap<DHTKey, BaseDirectoryInMemorySS> directories;
+  private Set<DHTKey> directoriesOnDiskAtBoot;
+  private File logDir;
+  private NamespaceOptions nsOptions;
+
+  private enum Mode {Eager, Lazy}
+
+  ;
+
+  public static String modeProperty = DirectoryServer.class.getCanonicalName() + ".Mode";
+  private static final Mode defaultMode = Mode.Lazy;
+  private static final Mode mode;
+
+  private static final boolean debug = false;
+  private static final boolean debugMergePut = false;
+
+  static FileDeletionWorker fileDeletionWorker = new FileDeletionWorker();
+
+  static {
+    String modeValue;
+    Mode _mode;
+
+    Log.warning("Initialized DirectoryServer class");
+    modeValue = PropertiesHelper.systemHelper.getString(modeProperty, UndefinedAction.ZeroOnUndefined);
+    if (modeValue == null) {
+      _mode = defaultMode;
+    } else {
+      try {
+        _mode = Mode.valueOf(modeValue);
+      } catch (Exception e) {
+        Log.logErrorWarning(e, "Using default mode " + defaultMode);
+        _mode = defaultMode;
+      }
+    }
+    mode = _mode;
+    Log.warningf("DirectoryServer mode %s", mode);
+  }
+
+  /*
+   * Note: DirectoryServer presently accepts all first stage put operations as valid, and eagerly merges in
+   * the changes observed. As a result, putUpdate() is a nop.
+   */
+
+  public DirectoryServer() {
+    directories = new ConcurrentHashMap<>();
+  }
+
+  @Override
+  public void initialize(SSNamespaceStore nsStore) {
+    this.nsStore = nsStore;
+    this.logDir = nsStore.getNamespaceSSDir();
+    this.nsOptions = nsStore.getNamespaceOptions();
+    directoriesOnDiskAtBoot = getDirectoriesOnDiskAtBoot();
+    new Persister();
+  }
+
+  private Set<DHTKey> getDirectoriesOnDiskAtBoot() {
+    Set<DHTKey> dirs;
+
+    dirs = new HashSet<>();
+    for (String s : logDir.list()) {
+      try {
+        if (!filterEmptyDir(s)) {
+          dirs.add(KeyUtil.keyStringToKey(s));
         }
-        mode = _mode;
-        Log.warningf("DirectoryServer mode %s", mode);
+      } catch (Exception e) {
+        Log.logErrorWarning(e);
+        Log.warningf("DirectoryServer.getDirectoriesOnDiskAtBoot() skipping %s", s);
+      }
     }
-    
-    /*
-     * Note: DirectoryServer presently accepts all first stage put operations as valid, and eagerly merges in
-     * the changes observed. As a result, putUpdate() is a nop.
-     */
-    
-    public DirectoryServer() {
-        directories = new ConcurrentHashMap<>();
+    return ImmutableSet.copyOf(dirs);
+  }
+
+  private boolean filterEmptyDir(String s) {
+    File f;
+
+    f = new File(logDir, s);
+    if (f.list().length == 0) {
+      Log.warningf("Removing empty dir: %s", s);
+      f.delete();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.